docs/auth.md

```diff
@@ -0,0 +1,96 @@
+## User Authentication in Soul
+
+Soul incorporates a robust user authentication system that handles user accounts, groups, permissions, and cookie-based user sessions. This section provides an overview of how the default implementation works.
+
+### Overview
+
+The Soul authentication system handles both authentication and authorization. Briefly, authentication verifies a user is who they claim to be, and authorization determines what an authenticated user is allowed to do. Here, the term "authentication" is used to refer to both tasks.
+
+The auth system consists of:
+
+- Users
+- Permissions: Binary (yes/no) flags designating whether a user may perform a certain task.
+- Default Permissions
+- Roles: A generic way of applying labels and permissions to more than one user.
+- A password hashing system
+- APIs for logging in users or restricting content
+
+The authentication system in Soul aims to be very generic and doesn't provide some features commonly found in web authentication systems, such as:
+
+- Password strength checking
+- Throttling of login attempts
+- Authentication against third-parties (OAuth, for example)
+- Object-level permissions
+
+#### Users
+
+User objects are the core of the authentication system. They typically represent the people interacting with your site and are used to enable things like restricting access, registering user profiles, associating content with creators, etc. Only one class of user exists in Soul's authentication framework, i.e., 'superusers' or types of users are just user objects with special attributes set, not different classes of user objects.
+
+The primary attributes of the default user are:
+
+- username
+- password
+- email
+- first_name
+- last_name
+
+Note that when Soul boots up, it looks for a table called "\_users" which holds the User objects mentioned above.
+
+Creating superusers:
+
+Create superusers using the createsuperuser command:
+
+```
+$ node src/server.js createsuperuser --username=joe --password=strongstring
+```
```

Comment:
Aah, are we providing this as a specific function in order to increase the security around super-users? (by preventing one from being created/modified via REST).  

If that's the case I think we need to prevent the `is_super_user` field being modified via `CREATE` or `UPDATE` calls.  I also think we need to provide `updatesuperuser` and `deletesuperuser` functions.

Reply:
Exactly you're right.

Reply:
Sure. So could you document the fact that it can't be edited by REST, and add these other maintenance functions please?

Reply:
Sure I will.